tests/units/cli/test_cli.py

```diff
@@ -575,6 +615,54 @@ def test_command_sync_tables(self):
         assert pytest_wrapped_e.type == SystemExit
         assert pytest_wrapped_e.value.code == 1
 
+    def test_command_sync_tables_cleanup_state_if_file_not_exists_and_no_tables_argument(self):
+        """Testing sync_tables cleanup state if file not exists and there is no tables argument"""
+        pipelinewise = self._init_for_sync_tables_states_cleanup()
+        self._assert_calling_sync_tables(pipelinewise)
+
+    def test_command_sync_tables_cleanup_state_if_file_not_exists_and_tables_argument(self):
+        """Testing sync_tables cleanup state if file not exists and there is tables argument"""
+        pipelinewise = self._init_for_sync_tables_states_cleanup(tables_arg='table1,table3')
+        self._assert_calling_sync_tables(pipelinewise)
+
+    def test_command_sync_tables_cleanup_state_if_file_exists_and_no_table_argument(self):
+        """Testing sync_tables cleanup state if file exists and there is no table argument"""
+        def _assert_state_file_is_deleted(*args, **kwargs):
+            # pylint: disable=unused-argument
+            assert os.path.isfile(test_state_file) is False
+
+        pipelinewise = self._init_for_sync_tables_states_cleanup()
+        test_state_file = pipelinewise.tap['files']['state']
+        self._make_sample_state_file(test_state_file)
+        self._assert_calling_sync_tables(pipelinewise, _assert_state_file_is_deleted)
```

Comment:
I don't understand why `_assert_state_file_is_deleted` is being used as the side effect of the `run_tap_fastsync` method

Reply:
we need to be sure that cleaning the state file is done before calling the `self.run_tap_fastsync` 
if we don't check this priority, and only check that file is cleaned in the test, test can be passed but in reality it's possible cleaning up the file be done after running this method (by bad implementation or mistake or updating the code in the future).
By having this side effect for the mocked method, It assures the file is already cleaned when the method is called! because it clean up is checked inside the side effect method.

Reply:
ok